### PR TITLE
Change preferred identity

### PR DESCRIPTION
**Pull request by leoapost**
### Context

- Ticket: N/A

This PR adds a new Admin page for super users only, to change the preferred identity of induction records.

### Changes proposed in this pull request
- Display a link in each Induction Record to change the preferred email
- Add a page to select the new email. All the emails listed are taken from the user's participant identities

### Guidance to review
1. Visit a ECF participant as a super user
2. Go to Induction Records tab
3. Click the `Change` link in the preferred email row
4. Confirm all the email in the dropdown emails are from the User's participant identities
5. Pick a new email and click `Submit`
6. Confirm the Induction Record's new preferred email

| Screenshots  |
|--------|
| image |
| image |

### CODE REVIEW

**Comment by ltello** [comment]
Nice one @leoapost. What if the user wants a brand new email?

**Review by ltello — APPROVED** [review]

File: app/controllers/admin/participants/induction_records_controller.rb [hunk 1]
```diff
@@ -13,10 +16,35 @@ def show
...
+    def edit_preferred_email
+      authorize @induction_record
+
+      @participant_identities = @participant_profile.user.participant_identities
+    end
+
+    def update_preferred_email
+      authorize @induction_record
+
+      if @induction_record.update(induction_params)
```
Comment (cwrw) [inline app/controllers/admin/participants/induction_records_controller.rb:28]: could we instead use the `Induction::ChangeInductionRecord` service to ensure this change creates a new induction record and the previous one is marked as changed??

Comment (leoapost) [reply app/controllers/admin/participants/induction_records_controller.rb:28]: @cwrw any particular reason we would need that?

**Comment by leoapost** [comment]
> Nice one @leoapost. What if the user wants a brand new email?

@ltello at the moment we add new emails manually. However, we could add an `New identity` action in the Identities tab in the Participant's admin page.

The scope of this PR is only to update the Induction records's preferred email.

File: app/controllers/admin/participants/induction_records_controller.rb (hunk 1 above)
Comment (cwrw) [reply app/controllers/admin/participants/induction_records_controller.rb:28]: I think usually induction records are audits to show the timeline of a profile, and if a status/school/provider/mentor id/schedule changes we create a new record and mark the previous as "changed". This also ensures the latest induction record is amended and not previous ones which might be out of date. Let me know your thoughts as to why not to do this

Comment (leoapost) [reply app/controllers/admin/participants/induction_records_controller.rb:28]: @cwrw oh right. I don't mind creating a new IR, but we already amend IRs directly (i.e. when we defer/withdraw a training) or when we fix IRs created with the wrong preferred email or when we assign mentors to participants. 

So, unless the IR's preferred email should be protected due to business rules I'm unaware, then I think it's ok to amend the IR.

**Pull request merged**